server/src/test/java/org/elasticsearch/indices/recovery/DanglingIndicesIT.java

```diff
@@ -126,29 +146,291 @@ public Settings onNodeStopped(String nodeName) throws Exception {
+     * being dangling.
+     */
+    public void testDanglingIndicesMustExistToBeImported() {
+        internalCluster().startNodes(1, buildSettings(0, true, false));
+
+        final ImportDanglingIndexRequest request = new ImportDanglingIndexRequest("NonExistentUUID", true);
+
+        boolean noExceptionThrown = false;
+        try {
+            client().admin().cluster().importDanglingIndex(request).actionGet();
+            noExceptionThrown = true;
+        } catch (Exception e) {
+            assertThat(e, instanceOf(IllegalArgumentException.class));
+            assertThat(e.getMessage(), containsString("No dangling index found for UUID [NonExistentUUID]"));
+        }
+
+        assertFalse("No exception thrown", noExceptionThrown);
+    }
+
+    /**
+     * Check that a dangling index can only be imported if "accept_data_loss" is set to true.
+     */
+    public void testMustAcceptDataLossToImportDanglingIndex() throws Exception {
+        internalCluster().startNodes(3, buildSettings(0, true, false));
+
+        final String stoppedNodeName = createDanglingIndices(INDEX_NAME);
+        final String danglingIndexUUID = findDanglingIndexForNode(stoppedNodeName, INDEX_NAME);
+
+        final ImportDanglingIndexRequest request = new ImportDanglingIndexRequest(danglingIndexUUID, false);
+
+        Exception caughtException = null;
+
+        try {
+            client().admin().cluster().importDanglingIndex(request).actionGet();
+        } catch (Exception e) {
+            caughtException = e;
+        }
+
+        assertNotNull("No exception thrown", caughtException);
+        assertThat(caughtException.getMessage(), containsString("accept_data_loss must be set to true"));
+    }
+
+    /**
+     * Check that dangling indices can be deleted. Since this requires that
+     * we add an entry to the index graveyard, the graveyard size must be
+     * greater than 1. To test deletes, we set the index graveyard size to
+     * 1, then create two indices and delete them both while one node in
+     * the cluster is stopped. The deletion of the second pushes the deletion
+     * of the first out of the graveyard. When the stopped node is resumed,
+     * only the second index will be found into the graveyard and the the
+     * other will be considered dangling, and can therefore be listed and
+     * deleted through the API
+     */
+    public void testDanglingIndexCanBeDeleted() throws Exception {
+        final Settings settings = buildSettings(1, true, false);
+        internalCluster().startNodes(3, settings);
+
+        final String stoppedNodeName = createDanglingIndices(INDEX_NAME, OTHER_INDEX_NAME);
+        final String danglingIndexUUID = findDanglingIndexForNode(stoppedNodeName, INDEX_NAME);
+
+        client().admin().cluster().deleteDanglingIndex(new DeleteDanglingIndexRequest(danglingIndexUUID, true)).actionGet();
+
+        assertThat(listDanglingIndices(), is(empty()));
```

Comment:
WDYT about making the same fix to this test as the REST test to verify that the index really did get deleted?